src/transformers/modeling_utils.py

```diff
@@ -419,6 +421,82 @@ def load(module: nn.Module, prefix=""):
     return error_msgs
 
 
+# a helper util to find the last sub-module and the param/buffer name
+def find_submodule_and_param_name(model, long_key, start_prefix):
+    # print(f"\n\n!!!{long_key} => {start_prefix}")
+
+    # XXX: Fixme
+    if len(start_prefix) > 0 and long_key.startswith(start_prefix):
+        long_key = ".".join(long_key.split(".")[1:])
+
+    split_key = long_key.split(".")
+    submodule = model
+    while len(split_key) > 1:
+        # print(f"{split_key[0]}")
+        if hasattr(submodule, split_key[0]):
+            # print(f"found {split_key[0]}")
+            submodule = getattr(submodule, split_key[0])
+            del split_key[0]
+        else:
+            # print(f"failed {split_key[0]}")
+            submodule = None
+            break
+    if submodule == model:
+        submodule = None
+    return submodule, split_key[0]
+
+
+def move_model_to_meta(model, loaded_state_dict_keys, start_prefix):
```

Comment:
The original `_load_pretrained_model_low_mem` hack got split into 2 functions, one that moves the model to meta and another that replaces the specific keys on meta to loaded state_dict keys.

that way I was able to integrate this functionality into the normal complex code of checking the keys and everything else.